File: conda/mpich/tests/helloworld.f90
program main

  use mpi
  implicit none

  integer :: provided,  ierr, size, rank, len
  character (len=MPI_MAX_PROCESSOR_NAME) :: name

  call MPI_Init(ierr)

  call MPI_Comm_rank(MPI_COMM_WORLD, rank, ierr)
  call MPI_Comm_size(MPI_COMM_WORLD, size, ierr)
  call MPI_Get_processor_name(name, len, ierr)

  write(*, '(2A,I2,A,I2,3A)') &
       'Hello, World! ', &
       'I am process ', rank, &
       ' of ', size, &
       ' on ', name(1:len), '.'

  call MPI_Finalize(ierr)

end program main

File: conda/mpich/tests/helloworld.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char *argv[])
{
  int provided, size, rank, len;
  char name[MPI_MAX_PROCESSOR_NAME];

  MPI_Init(&argc, &argv);

  MPI_Comm_size(MPI_COMM_WORLD, &size);
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
  MPI_Get_processor_name(name, &len);

  printf("Hello, World! I am process %d of %d on %s.\n", rank, size, name);

  MPI_Finalize();
  return 0;
}
